plugins/restful/RestfulDataProviderVariable.php

```diff
@@ -0,0 +1,306 @@
+<?php
+
+/**
+ * @file
+ * Contains \RestfulDataProviderDbQuery
+ */
+
+abstract class RestfulDataProviderVariable extends \RestfulBase implements \RestfulDataProviderInterface {
```

Comment:
Maybe we need to create a `RestfulDataProviderVariableInterface`. Just like we have `RestfulDataProviderCToolsPluginsInterface`.